### PR TITLE
Feature/recipe repository class

### LINKED ISSUE #23: RecipeRepository METHOD:  filtered list of recipes based on one or more tags.
As a developer I need a 
METHOD to  filter a list of recipes based on one or more tags.

### LINKED ISSUE #24: RecipeRepository METHOD: filtered list of recipes based on its name or ingredients.
As a dev, I need a METHOD to filter list of recipes based on its name or ingredients.

### PR DESCRIPTION
**Pull request by darlaevans2000**
# Description
Add all methods needed for RecipeRepository class and all needed testing as well. 

closes #23
closes #24

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
npm test, chai, mocha

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "html-webpack-plugin": "^4.5.1",
     "mocha": "^8.2.1",
     "mochapack": "^2.0.6",
+    "prettier": "2.3.0",
```
Comment (clairefields15) [inline package.json:14]: Nice! Glad you were able to get this working

File: src/classes/RecipeRepository.js [hunk 2]
```diff
@@ -8,10 +8,45 @@ class RecipeRepository {
...
+    }, [])
+    return filteredRecipes;
   }
-}
 
-export default RecipeRepository;
+  filterByIngredients(ingredients) {
+    const filteredRecipes = ingredients.reduce((acc, ingredient) => {
+      this.recipesData.forEach(recipe => {
+        const recipeIngredients = recipe.getIngredientNames();
```
Comment (clairefields15) [inline src/classes/RecipeRepository.js:26]: Great use of our existing method

File: src/classes/RecipeRepository.js [hunk 3]
```diff
@@ -8,10 +8,45 @@ class RecipeRepository {
...
+        if(recipeIngredients.includes(ingredient) && !acc.includes(ingredient)) {
+          acc.push(recipe)
+        }
+      });
+      return acc
+    }, []);
+    return filteredRecipes;
+  }
+
+  filterByName(names) {
```
Comment (clairefields15) [inline src/classes/RecipeRepository.js:36]: NIce

File: test/RecipeRepository-test.js [hunk 4]
```diff
@@ -24,15 +24,38 @@ describe('RecipeRepository', () => {
...
+  it('Should be able to filter recipes by any name', () => {
+    recipe1 = new Recipe(recipeData[0]);
+    recipe2 = new Recipe(recipeData[1]);
+    const data = [recipe1, recipe2]
+    recipeRepository = new RecipeRepository(data);
+
+    
+    const recipesByName = recipeRepository.filterByName(['Loaded Chocolate Chip Pudding Cookie Cups']);
+
+    expect(recipesByName).to.deep.equal([recipe1])
```
Comment (clairefields15) [inline test/RecipeRepository-test.js:58]: Love it! We will definitely have to add some sad path testing here in the future, to test for edge cases in user input (capital letters, double spaces, weird characters, etc)

**Review by clairefields15 — APPROVED** [review]
Looks great, we'll just have to add some sad path testing to filter recipes by name/ingredient in the future. No sad path testing needed for tags as that will be handled by our UI.
